imagebuildah/build.go

```diff
@@ -1200,6 +1244,12 @@ func (b *Executor) Build(ctx context.Context, stages imagebuilder.Stages) error
 			return errors.Errorf("Failed to cleanup intermediate containers")
 		}
 	}
+	// Remove intermediate images that we created for AS clause handling
+	for _, value := range imageMap {
+		if err := b.store.Delete(value); err != nil {
+			logrus.Debugf("unable to remove intermediate image %q: %v", value, err)
+		}
+	}
```

Comment:
@nalind I initially used b.store.DeleteImage(value, true) at line 1249, however it always failed into the Debugf statement and left the extra image behind.  I think it's safe to delete it outright like this, but wanted to verify with you.

Reply:
What was the error that `DeleteImage()` was returning?  Plain `Delete()` skips over some safety checks that we might want to not skip.

Reply:
"image is in use by a container" is the error.  For a test, I replaced the debug statement with a printf and prepended "TOM:" to it to make it stand out.  I'm not sure where the container/image is that it's complaining about, I don't see it in the output nor in the 'buildah images/containers output'.   I thought I had seen it yesterday though.


Test output:

```
# buildah bud -t tom -f ~/Dockerfile.fromas .
STEP 1: FROM centos:7 AS base
Getting image source signatures
Copying blob sha256:256b176beaff7815db2a93ee2071621ae88f451bb1e198ca73010ed5bba79b65
 71.23 MiB / 71.23 MiB [===================================================] 10s
Copying config sha256:5182e96772bf11f4b912658e265dfe0db8bd314475443b6434ea708784192892
 2.13 KiB / 2.13 KiB [======================================================] 0s
Writing manifest to image destination
Storing signatures
STEP 2: RUN touch /1
STEP 3: ENV LOCAL=/1
STEP 4: RUN find $LOCAL
/1
STEP 5: COMMIT containers-storage:[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.override_kernel_check=true]localhost/tom:latest
Getting image source signatures
Skipping fetch of repeat blob sha256:1d31b5806ba40b5f67bde96f18a181668348934a44c9253b420d5f04cfb4e37a
Copying blob sha256:a034a2927de80dc639e5e5f0eece366629321a3cbee05c2e49e24af6fe49443a
 167 B / 167 B [============================================================] 0s
Copying config sha256:821f92550dcdef1d20bc9f5cddc2fc2d8a04197592a5c062ed0ce9a8fdc3a88e
 1.18 KiB / 1.18 KiB [======================================================] 0s
Writing manifest to image destination
Storing signatures
--> 821f92550dcdef1d20bc9f5cddc2fc2d8a04197592a5c062ed0ce9a8fdc3a88e
STEP 6: FROM base
STEP 7: RUN find $LOCAL
/1
STEP 8: COMMIT containers-storage:[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.override_kernel_check=true]localhost/tom:latest
Getting image source signatures
Skipping fetch of repeat blob sha256:1d31b5806ba40b5f67bde96f18a181668348934a44c9253b420d5f04cfb4e37a
Skipping fetch of repeat blob sha256:d2f84891591559baf3edea8db4273078145c71ba4d590d0a8368984a66afa42c
Copying blob sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1
 32 B / 32 B [==============================================================] 0s
Copying config sha256:46d9491b593562217acc42482afc9f74ffa57fc471d982eda43755ef1fe9db44
 1.29 KiB / 1.29 KiB [======================================================] 0s
Writing manifest to image destination
Storing signatures
--> 46d9491b593562217acc42482afc9f74ffa57fc471d982eda43755ef1fe9db44
TOM: unable to remove intermediate image 821f92550dcdef1d20bc9f5cddc2fc2d8a04197592a5c062ed0ce9a8fdc3a88e: Image used by 3a07f584e4de06e1e9c7536caea12c52c1d270e86c19b0dcc17f3c2756ae3c13: image is in use by a container

```

Reply:
For the purpose of this PR then, should I stick with DeleteImage or the straight Delete call?

Reply:
`DeleteImage` is heavily preferred - `Delete` removes the record of the image, but not its layers.

Reply:
That's what I thought, but thought I'd just double check as using it means we'll likely leave an image behind.